File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/TarifaEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class TarifaEntity : BaseEntityObject
    {
        public TarifaEntity()
        {
            this.TarifaId = GetInt32Default();
            this.UnidadMedidaId = GetInt32Default();
            this.MonedaId = GetInt32Default();
            this.PorcentajeImpuestoId = GetInt32Default();
            this.PrecioSinInpuesto = GetInt32Default();
            this.PrecioConInpuesto = GetInt32Default();
            this.FechaCreacion = DateTime.Now;
            this.Vigente = GetBooleanDefault();
            this.ObjetoReferenciaId = GetInt32Default();
            this.MercaderiaId = GetInt32Default();
        }

        public Int32 TarifaId { get; set; }
        public Int32 UnidadMedidaId { get; set; }
        public Int32 MonedaId { get; set; }
        public Int32 PorcentajeImpuestoId { get; set; }
        public Decimal PrecioSinInpuesto { get; set; }
        public Decimal PrecioConInpuesto { get; set; }
        public DateTime FechaCreacion { get; set; }
        public Boolean Vigente { get; set; }

        public Int32 ObjetoReferenciaId { get; }
        public Int32 MercaderiaId { get; set; }


    }
}

File: ProyectoMysql/server/EntityLayer/TarifaEntity.py
from datetime import datetime
from pydantic import BaseModel,validator
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum

class TarifaMainModel(BaseModel):
    TarifaId: int = 0
    NomProducto: str = ''
    NomUnidad: str = ''
    NomMoneda: str = ''
    Valor: str = ''
    PrecioSinInpuesto: float = 0
    PrecioConInpuesto: float = 0
    FechaCreacion: datetime = datetime.now()
    Vigente: bool = False
    
    @validator('Vigente', pre=True)
    def convertir_a_bool(cls, v: any) -> bool:
        if isinstance(v, str):
            if v.lower() in ('true', '1'):
                return True
            elif v.lower() in ('false', '0'):
                return False
        return bool(v)
    
    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)

class TarifaSaveModel(BaseModel):
    TarifaId: int = 0
    MercaderiaId: int = 0
    MonedaId: int = 0
    UnidadMedidaId: int = 0
    PorcentajeImpuestoId: int = 0
    PrecioSinInpuesto: float = 0
    PrecioConInpuesto: float = 0
    FechaCreacion: datetime = datetime.now()
    Vigente: bool = False
    
    @validator('Vigente', pre=True)
    def convertir_a_bool(cls, v: any) -> bool:
        if isinstance(v, str):
            if v.lower() in ('true', '1'):
                return True
            elif v.lower() in ('false', '0'):
                return False
        return bool(v)
    
    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)